include/bootchooser.h

```diff
@@ -20,7 +20,8 @@ GQuark r_bootchooser_error_quark(void);
  *
  * @return TRUE if successful, FALSE if failed
  */
-gboolean r_boot_set_state(RaucSlot *slot, gboolean good, GError **error);
+gboolean r_boot_set_state(RaucSlot *slot, gboolean good, GError **error)
+G_GNUC_WARN_UNUSED_RESULT;
```

Comment:
I agree this is useful to prevent unintentionally ignoring failures in methods and indirectly enforce error checking a bit stricter. I would love to see a line in the commit message about the value added by these changes to give other developers a good hint. Please also note that you adapted the test functions in order to fix compile errors now.

Reply:
OK.